src/dbg/enumhandles.cpp

```diff
@@ -0,0 +1,122 @@
+#include "_global.h"
+#include "debugger.h"
+#include "TitanEngine\TitanEngine.h"
+
+struct SYSTEM_HANDLE_INFORMATION{
+    ULONG ProcessId;
+    UCHAR ObjectTypeNumber;
+    UCHAR Flags;
+    USHORT Handle;
+    PVOID Object;
+    DWORD GrantedAccess;
+};
+
+struct OBJECT_TYPE_INFORMATION
+{
+    UNICODE_STRING Name;
+    ULONG TotalNumberOfObjects;
+    ULONG TotalNumberOfHandles;
+    ULONG TotalPagedPoolUsage;
+    ULONG TotalNonPagedPoolUsage;
+    ULONG TotalNamePoolUsage;
+    ULONG TotalHandleTableUsage;
+    ULONG HighWaterNumberOfObjects;
+    ULONG HighWaterNumberOfHandles;
+    ULONG HighWaterPagedPoolUsage;
+    ULONG HighWaterNonPagedPoolUsage;
+    ULONG HighWaterNamePoolUsage;
+    ULONG HighWaterHandleTableUsage;
+    ULONG InvalidAttributes;
+    GENERIC_MAPPING GenericMapping;
+    ULONG ValidAccess;
+    BOOLEAN SecurityRequired;
+    BOOLEAN MaintainHandleCount;
+    USHORT MaintainTypeList;
+    DWORD PoolType;
+    ULONG PagedPoolUsage;
+    ULONG NonPagedPoolUsage;
+};
+
+struct MYHANDLES{
+    DWORD_PTR HandleCount;
+    SYSTEM_HANDLE_INFORMATION Handles[1];
+};
+
+#ifdef _WIN64
+DWORD (*NtQuerySystemInformation)(DWORD SystemInfoClass, void* SystemInfo, DWORD SystemInfoSize, DWORD* ReturnedSize) = nullptr;
+#else //x86
+DWORD(__stdcall *NtQuerySystemInformation)(DWORD SystemInfoClass, void* SystemInfo, DWORD SystemInfoSize, DWORD* ReturnedSize) = nullptr;
+#endif //_WIN64
+#ifdef _WIN64
+DWORD (*NtQueryObject)(HANDLE ObjectHandle, ULONG ObjectInformationClass, PVOID ObjectInformation, ULONG ObjectInformationLength, PULONG ReturnLength) = nullptr;
+#else //x86
+DWORD(__stdcall *NtQueryObject)(HANDLE ObjectHandle, ULONG ObjectInformationClass, PVOID ObjectInformation, ULONG ObjectInformationLength, PULONG ReturnLength) = nullptr;
+#endif //_WIN64
+
+extern "C" DLL_EXPORT long _dbg_enumhandles(duint* handles, unsigned char* typeNumbers, unsigned int* grantedAccess, unsigned int maxcount)
+{
+    MYHANDLES* myhandles = (MYHANDLES*)emalloc(16384, "_dbg_enumhandles");
```

Comment:
For this you can use `Memory<MYHANDLES*> myhandles(16384)` to avoid the possibility of memory leaks.